docs: fix info regarding is_global entry in .globalconfig

kriwar opened the pull request:
## Summary

The note regarding the necessity of `is_global = true` in files named `.globalconfig` is incorrect since https://github.com/dotnet/roslyn/pull/49834.

gewarren commented on docs/fundamentals/code-analysis/configuration-files.md:
```diff
@@ -97,7 +97,7 @@ Consider the following naming recommendations:
 
 > [!NOTE]
-> The top-level entry `is_global = true` is required even when the file is named `.globalconfig`.
+> The top-level entry `is_global = true` is not required when the file is named `.globalconfig`, but it is recommended to include for clarity.
```
```suggestion
> The top-level entry `is_global = true` is not required when the file is named `.globalconfig`, but it is recommended for clarity.
```

kriwar replied:
Changed wording as suggested (amended).

I'm not familiar with the internals of dotnet releases. But in the dotnet PR I linked, the change was merged by @chsienki to a branch called `release/dev16.9` on 19 Dec 2020.

gewarren reviewed (approved):
It would be good to add "since version xyz" to the statement, if you know that info.

kriwar committed:
docs: fix info regarding is_global entry in .globalconfig

The note regarding the necessity of `is_global = true` in files named `.globalconfig` is incorrect since https://github.com/dotnet/roslyn/pull/49834.

gewarren commented:
Thanks @kriwar!
